omod/src/main/java/org/openmrs/module/metrics/util/MetricHandler.java

```diff
@@ -0,0 +1,69 @@
+package org.openmrs.module.metrics.util;
+
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+import java.lang.management.ManagementFactory;
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.jmx.JmxReporter;
+import org.openmrs.module.metrics.api.exceptions.MetricsException;
+import org.openmrs.module.metrics.api.service.MetricService;
+import org.openmrs.module.metrics.api.utils.EventsUtils;
+import org.openmrs.module.metrics.builder.JmxReportBuilder;
+import org.openmrs.module.metrics.model.MetricscConfigImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class MetricHandler {
+	
+	private static final Logger LOGGER = LoggerFactory.getLogger(MetricHandler.class);
+	
+	@Autowired
+	private JmxReportBuilder jmxReportBuilder;
+	
+	@Autowired
+	private MetricService metricService;
+	
+	public void buildMetricFlow(LocalDateTime startRange, LocalDateTime endRange) throws MetricsException {
+		
+		//fetch custom metrics
+		Integer noOfNewPatients = metricService.getNewPatientsObjectsByGivenDateRange(startRange, endRange);
+		Map<String, Integer> noOfEncounters = metricService.getEncounterObjectTypesCountByGivenDateRange(startRange,
+		    endRange);
```

Comment:
These would seem to be better implemented as [Gauges or Histograms](https://metrics.dropwizard.io/3.1.0/manual/core/#man-core-counters), so that they can update dynamically, rather than simply being static values.